src/SourceBuild/content/test/Microsoft.DotNet.Tests/PackageSourceMappingsTests.cs

```diff
@@ -0,0 +1,242 @@
+        [Fact]
+        public void UnifiedBuildWithMappings()
+        {
+            string[] sources = ["source-built-arcade", "source-built-runtime", "net-sdk-supporting-feed"];
+            RunTest("ub-mappings.config", true, sources, customSources: ["net-sdk-supporting-feed"], sourceBuild: false);
+        }
+
+        // Unified build - with mappings - online - no local sources
+        [Fact]
+        public void UnifiedBuildWithMappingsNoLocalSources()
+        {
+            string[] sources = ["net-sdk-supporting-feed"];
+            RunTest("ub-mappings-nolocal.config", true, sources, customSources: ["net-sdk-supporting-feed"], sourceBuild: false);
+        }
+
+        // Unified build - no mappings - online
+        [Fact]
+        public void UnifiedBuildNoMappings()
+        {
+            string[] sources = ["source-built-arcade", "source-built-runtime", "net-sdk-supporting-feed"];
+            RunTest("ub.config", true, sources, customSources: ["net-sdk-supporting-feed"], sourceBuild: false);
+        }
+
+        // Source build tests - with and without mappings - online and offline
+        [Theory]
+        [InlineData("sb-online.config", true)]
+        [InlineData("sb-offline.config", false)]
+        [InlineData("sb-mappings-online.config", true)]
+        [InlineData("sb-mappings-offline.config", false)]
+        public void SourceBuildTests(string nugetConfigFilename, bool useOnlineFeeds)
+        {
+            string[] sources = ["prebuilt", "previously-source-built", "reference-packages",
+                                "source-built-arcade", "source-built-runtime"];
+            RunTest(nugetConfigFilename, useOnlineFeeds, sources);
+        }
+
+        // Source build - SBRP repo - online and offline
+        [Theory]
+        [InlineData("sb-sbrp-offline.config", false)]
+        [InlineData("sb-sbrp-online.config", true)]
+        public void SourceBuildSbrpRepoTests(string nugetConfigFilename, bool useOnlineFeeds)
+        {
+            string[] sources = ["prebuilt", "previously-source-built", "reference-packages"];
+            RunTest(nugetConfigFilename, useOnlineFeeds, sources);
+        }
+
+        private static void RunTest(string nugetConfigFilename, bool useOnlineFeeds, string[] sources, string[]? customSources = null, bool sourceBuild = true)
+        {
+            string psmAssetsDir = Path.Combine(Directory.GetCurrentDirectory(), "assets", "PackageSourceMappingsTests");
+            string originalNugetConfig = Path.Combine(psmAssetsDir, "original", nugetConfigFilename);
+            string expectedNugetConfig = Path.Combine(psmAssetsDir, "expected", nugetConfigFilename);
+
+            string modifiedNugetConfig = Path.Combine(PackageSourceMappingsSetup.PackageSourceMappingsRoot, nugetConfigFilename);
+            Directory.CreateDirectory(Path.GetDirectoryName(modifiedNugetConfig)!);
+            File.Copy(originalNugetConfig, modifiedNugetConfig, true);
```

Comment:
Are we doing all this in the source tree? Tests shouldn't mutate the source tree. Instead we usually call the create temp directory API to get a temp file / folder.

Reply:
Are you sure? Isn't this under `cwd/PackageSourceMappingsTests`?

Reply:
`cwd` is the app's root, i.e. `artifacts/bin/Microsoft.DotNet.Tests/Release/`

Reply:
Right and that's not a location that we should write to, if I'm not mistaken. In runtime all our tests use a temp location. @jkotas probably knows more about the why.

Reply:
It's an easy fix, so I'll update this.

Reply:
Fixed with https://github.com/dotnet/sdk/pull/44649/commits/6337002902416768d8776d865625b83fd6d75d79